azure-pipelines.yml

```diff
@@ -74,8 +74,9 @@ stages:
 
           - bash: |
               python -m pip install --upgrade pip
-              python -m pip install --upgrade -r requirements/dev/build-requirements.txt
-              python -m pip install -r requirements/testing/all.txt -r requirements/testing/extra.txt
+              python -m pip install --upgrade --group build
+              python -m pip install --group test
+              python -m pip install --prefer-binary --group test-extra
```

Comment:
Worth a comment explaining the reason for using `--prefer-binary` just for `test-extra`?

Reply:
This came from the original file: https://github.com/matplotlib/matplotlib/pull/29281/files#diff-1823d2c36efddd45c3c2fa83d0fa74b23c3559ee9ceb371d4dfcb14372d9efaeL3

But I'm actually unsure if it's applied to only the requirements file that contained it, or the entire command-line.